python/pyspark/pandas/internal.py

```diff
@@ -616,8 +628,7 @@ def __init__(
         >>> internal.column_label_names
         [('column_labels_a',), ('column_labels_b',)]
         """
-
-        assert isinstance(spark_frame, SparkDataFrame)
+        assert isinstance(spark_frame, (LegacyDataFrame, ConnectDataFrame))
```

Comment:
`isinstance(spark_frame, GenericDataFrame.__args__)` should work?

Reply:
Nice workaround!